pkg/kubectl/describe.go

```diff
@@ -856,14 +856,26 @@ func describeContainers(containers []api.Container, containerStatuses []api.Cont
 		}
 		fmt.Fprintf(out, "    Environment Variables:%s\n", none)
 		for _, e := range container.Env {
-			if e.ValueFrom != nil && e.ValueFrom.FieldRef != nil {
-				var valueFrom string
-				if resolverFn != nil {
-					valueFrom = resolverFn(e)
-				}
-				fmt.Fprintf(out, "      %s:\t%s (%s:%s)\n", e.Name, valueFrom, e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath)
-			} else {
+			if e.ValueFrom == nil {
 				fmt.Fprintf(out, "      %s:\t%s\n", e.Name, e.Value)
+			} else {
+				if e.ValueFrom.FieldRef != nil {
+					var valueFrom string
+					if resolverFn != nil {
+						valueFrom = resolverFn(e)
+					}
+					fmt.Fprintf(out, "      %s:\t%s (%s:%s)\n", e.Name, valueFrom, e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath)
+					continue
+				}
+
+				if e.ValueFrom.SecretKeyRef != nil {
+					fmt.Fprintf(out, "      %s:\t<set to the key '%s' in secret '%s'>\n", e.Name, e.ValueFrom.SecretKeyRef.Key, e.ValueFrom.SecretKeyRef.Name)
+					continue
+				}
+
+				if e.ValueFrom.ConfigMapKeyRef != nil {
+					fmt.Fprintf(out, "      %s:\t<set to the key '%s' of configMap '%s'>\n", e.Name, e.ValueFrom.ConfigMapKeyRef.Key, e.ValueFrom.ConfigMapKeyRef.Name)
```

Comment:
@pmorie shot the idea down himself after proposing it :)
https://github.com/kubernetes/kubernetes/issues/23679#issuecomment-204056532
We had the same debate in the console, and ultimately decided no for the
same reasons

On Wed, Apr 6, 2016 at 9:04 AM, David Eads <notifications@github.com> wrote:

> In pkg/kubectl/describe.go
> <https://github.com/kubernetes/kubernetes/pull/23890#discussion_r58700144>
> :
>
> > +				if e.ValueFrom.FieldRef != nil {
> > +					var valueFrom string
> > +					if resolverFn != nil {
> > +						valueFrom = resolverFn(e)
> > +					}
> > +					fmt.Fprintf(out, "      %s:\t%s (%s:%s)\n", e.Name, valueFrom, e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath)
> > +					continue
> > +				}
> > +
> > +				if e.ValueFrom.SecretKeyRef != nil {
> > +					fmt.Fprintf(out, "      %s:\t<set to the key '%s' in secret '%s'>\n", e.Name, e.ValueFrom.SecretKeyRef.Key, e.ValueFrom.SecretKeyRef.Name)
> > +					continue
> > +				}
> > +
> > +				if e.ValueFrom.ConfigMapKeyRef != nil {
> > +					fmt.Fprintf(out, "      %s:\t<set to the key '%s' of configMap '%s'>\n", e.Name, e.ValueFrom.ConfigMapKeyRef.Key, e.ValueFrom.ConfigMapKeyRef.Name)
>
> Can you retrieve the actual value here? It shouldn't be sensitive and it
> would help people know if they got their expression right.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23890/files/648a199b9dd006a6acaf085a4a5d2abcf7ed1ffa#r58700144>
>